src/askui/models/anthropic/computer_agent.py

```diff
@@ -30,16 +35,23 @@ def __init__(
     def _create_message(
         self, messages: list[MessageParam], model_choice: str
     ) -> MessageParam:
+        _thinking: ThinkingConfigParam = ThinkingConfigDisabledParam(type="disabled")
+
+        if self._settings.thinking:
+            _thinking = ThinkingConfigEnabledParam(
+                type="enabled",
+                budget_tokens=self._settings.thinking.budget_tokens,
+            )
+
         response = self._client.beta.messages.with_raw_response.create(
             max_tokens=self._settings.max_tokens,
-            messages=[
-                cast("BetaMessageParam", message.model_dump(mode="json"))
-                for message in messages
-            ],
+            messages=cast("list[BetaMessageParam]", messages),
```

Comment:
This seems incorrect to me. We are casting a list of pydantic.BaseModels to a list of dictionaries... So what was the reasoning behind this?